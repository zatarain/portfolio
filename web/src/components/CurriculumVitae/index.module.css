.curriculum {
	font-size: 1rem;
}

.curriculum h2 {
	font-size: 1.75rem;
	font-family: var(--headings-font);
	color: var(--heading-2);
	margin: 1rem 0 0.5rem;
}

.curriculum h3 {
	font-size: 1.2rem;
	font-family: var(--headings-font);
	color: var(--heading-3);
	margin: 0;
}

.curriculum ::selection {
	color: black;
	background-color: var(--ldn-lighter);
}

.curriculum dd {
	padding: 0;
	margin: 0;
}

.experiences ul {
	list-style-type: none;
	padding: 0;
	margin: 0;
}

.experiences li {
	display: grid;
	grid-gap: 0 2rem;
	grid-template-columns: 10rem auto;
	margin: 1rem 0;
}

.experiences h3 {
	grid-column: 1 / span 2;
}

.experiences dl {
	text-align: center;
}

.experiences .contributions ul {
	margin-top: -1rem;
	padding-left: 1.5rem;
}

.experiences .contributions li {
	display: list-item;
	margin: 0;
}

.experiences dt {
	display: none;
}

.education > ul {
	list-style-type: none;
	display: grid;
	grid-template-columns: 1fr 1fr;
	gap: 0.7rem;
	margin: 0;
	padding: 0;
}

.education p {
	margin: 0;
}

.education dl {
	margin: 0;
}

.education dt {
	display: inline-block;
	margin-right: 1ch;
}

.education dd {
	display: inline-block;
}

.education .subjects {
	display: block;
}

.ticks {
	padding-left: 1.5rem;
}

@media screen {
	.ticks li::before, .experiences .contributions li::before {
		content: '';
		display: inline-block;
		background: url('/tick.svg') no-repeat;
		background-position: 0 .15rem;
		width: 1rem;
		height: 1rem;
		background-size: 1rem;
		position: absolute;
		left: -1.25rem;
		top: 0.25rem;
	}
}

.ticks li {
	list-style: none;
	position: relative;
}

.experiences .contributions li {
	position: relative;
}

@media screen and (max-width: 800px) {
	.experiences dl {
		display: inline;
		grid-area: unset;
		width: auto;
		text-align: left;
		margin: 0;
	}

	.experiences dl::before {
		content: '(';
	}

	.experiences dl::after {
		content: ')';
	}

	.experiences dt {
		display: inline-block;
		margin: 0 0.25rch;
		font-weight: 500;
	}

	.experiences dd {
		display: inline-block;
		margin: 0 0.25rch;
		font-style: italic;
	}

	.experiences li {
		display: block;
		grid-template-areas: none;
	}

	.education > ul {
		display: block;
	}
}

@media print {
	.curriculum {
		font-size: 10pt;
	}

	.curriculum h2 {
		font-size: 1.2em;
		font-family: var(--headings-font);
		color: var(--heading-2);
		margin: 0.5em 0 0.05em;
	}

	.curriculum h3 {
		font-size: 1em;
		font-family: var(--headings-font);
		color: var(--heading-3);
		margin: 0.2em 0 0;
	}

	.curriculum p {
		margin: 0.1em 0;
		font-size: 0.95em;
	}

	.curriculum dl {
		margin: 0.1em 0;
	}

	.curriculum li {
		margin: 0.1em 0;
		font-size: 0.95em;
	}

	.ticks li {
		list-style-type: disc;
	}

	.experiences li {
		grid-gap: 0 .5rem;
		grid-template-columns: 7rem auto;
		page-break-inside: avoid;
	}

	.experiences .contributions li {
		list-style-type: disc;
	}

	.experiences dl {
		font-size: 0.8em;
		width: 9.5em;
	}

	.education > ul {
		gap: 0;
	}
}
