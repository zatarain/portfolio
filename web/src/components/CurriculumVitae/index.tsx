import Markdown from '#components/Markdown'
import styles from './index.module.css'
export interface Job {
	role: string
	type: string
	company: string
	city: string
	country: string
	start: string
	end: string
	contributions: string
}

export interface Education {
	school: string
	grade: string
	thesis?: string
	start: number
	end: number
	gpa: number
	subjects?: string[]
}

export interface AcademicProject {
	title: string
	type: string
	start?: string
	end?: string
	duration?: string
	description: string
}

export interface CurriculumVitaeProperties {
	name: string
	email?: string[]
	phone?: string[]
	social?: object[]
	website?: string
	statement?: string
	experience?: Job[]
	education?: Education[]
	projects?: AcademicProject[]
	skills?: object[]
	awards?: string[]
	leadership?: string
	volunteering?: string
	tex?: string
}

interface Properties {
	data: CurriculumVitaeProperties
}

const CurriculumVitae = ({ data }: Properties) => {
	return (
		<div className={styles.curriculum}>
			<section id="personal-statement" className={styles.statement}>
				<h2>Who am I?</h2>
				<p>{data.statement}</p>
			</section>
			<section id="work-experience" className={styles.experiences}>
				<h2>Work Experience</h2>
				<ul>
					{data.experience?.map((job: Job, index: number) =>
						<li key={index}>
							<h3>{`${job.role} @ ${job.company} - ${job.city}, ${job.country}`}</h3>
							<dl>
								<dt>Type:</dt><dd>{job.type}</dd>
								<dt>Duration:</dt><dd>{`${job.start} - ${job.end}`}</dd>
							</dl>
							<Markdown content={job.contributions} />
						</li>
					)}
				</ul>
			</section>
			<section id="education" className={styles.education}>
				<h2>Education</h2>
				<ul>
					{data.education?.map((education: Education, index: number) =>
						<li key={index}>
							<h3>{`${education.school}`}</h3>
							<p>{`${education.grade} (${education.start} - ${education.end})`}</p>
							<dl>
								{education.thesis && <><dt>Thesis:</dt><dd>{education.thesis}</dd></>}
								<dt>Cumulative GPA:</dt><dd>{education.gpa}/100</dd>
								<dt className={styles.subjects}>Most relevant subjects and courses:</dt>
								<dd><ul>{education.subjects?.map((subject, key) => <li key={key}>{subject}</li>)}</ul></dd>
							</dl>
						</li>
					)}
				</ul>
			</section>
			<section id="academic-projects" className={styles.experiences}>
				<h2>Major Academic Projects</h2>
				<ul>
					{data.projects?.map((project: AcademicProject, index: number) =>
						<li key={index}>
							<h3>{project.title}</h3>
							<dl>
								<dt>Type:</dt><dd>{project.type}</dd>
								<dt>Duration:</dt><dd>{project.duration && project.duration || `${project.start} - ${project.end}`}</dd>
							</dl>
							<Markdown content={project.description} />
						</li>
					)}
				</ul>
			</section>
			<section id="awards">
				<h2>Awards</h2>
				<ul>
					{data.awards?.map((award: string, index: number) => <li key={index}>{award}</li>)}
				</ul>
			</section>
			<section id="leadership">
				<h2>Leadership</h2>
				<Markdown content={data.leadership} />
			</section>
			<section id="volunteering">
				<h2>Volunteering Activities</h2>
				<Markdown content={data.volunteering} />
			</section>
			<section>
				<h2>TeX</h2>
				<Markdown content={data.tex} />
			</section>
		</div>
	)
}

export default CurriculumVitae
