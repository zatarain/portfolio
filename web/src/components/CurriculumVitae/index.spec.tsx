import { render } from '@testing-library/react'
import CurriculumVitae from '.'

describe('<CurriculumVitae />', () => {
	it('renders the component', () => {
		const data = {
			name: 'My Test Name',
			statement: 'Test statement paragraph',
			education: [
				{
					school: 'dummy school',
					grade: 'BSc Dummy Career',
					start: 2019,
					end: 2021,
					gpa: 90,
					subjects: [
						'hello',
						'world',
					],
				},
				{
					school: 'dummy research centre',
					grade: 'MSc Dummy Career',
					thesis: 'Dummy Thesis Topic',
					start: 2021,
					end: 2023,
					gpa: 95,
					subjects: [
						'hello master',
						'world master',
					],
				},
			],
			experience: [
				{
					role: 'dummy role',
					type: 'full-time',
					company: 'dummy company',
					city: 'dummy city',
					country: 'dummy country',
					start: '10/2021',
					end: '10/2022',
					contributions: 'dummy contributions paragraph',
				},
				{
					role: 'dummy role intern',
					type: 'part-time',
					company: 'same company',
					city: 'another city',
					country: 'same country',
					start: '06/2021',
					end: '09/2021',
					contributions: 'dummy contributions paragraph as intern',
				},
			],
			projects: [
				{
					title: 'Dummy Project 01',
					type: 'thesis',
					start: '08/2013',
					end: '11/2014',
					description: 'Dummy description paragraph 01',
				},
				{
					title: 'Dummy Project 02',
					type: 'personal',
					duration: 'Spring 2023',
					description: 'Dummy description paragraph 02',
				},
			],
			awards: [
				'Dummy Award 01',
				'Dummy Award 02',
			],
			leadership: 'Test leadership paragraph',
			volunteering: 'Test volunteering paragraph',
			pictures: [
				{
					id: '1001',
					media_url: 'https://cdn.instagram.com/dummy-01.jpg',
					caption: 'Dummy caption 01',
				},
				{
					id: '1001',
					media_url: 'https://cdn.instagram.com/dummy-01.jpg',
					caption: 'Dummy caption 01',
				},
			],
		}
		const { asFragment, getByText } = render(<CurriculumVitae data={data} />)

		expect(getByText('Test statement paragraph')).toBeInTheDocument()
		expect(asFragment()).toMatchSnapshot()
	})
})
