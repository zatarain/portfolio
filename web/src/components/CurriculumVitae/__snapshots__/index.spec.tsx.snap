// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurriculumVitae /> renders the component 1`] = `
<DocumentFragment>
  <div
    class="curriculum"
  >
    <section
      id="personal-statement"
    >
      <h2>
        Who am I?
      </h2>
      <p>
        Test statement paragraph
      </p>
    </section>
    <section
      class="experiences"
      id="work-experience"
    >
      <h2>
        Work Experience
      </h2>
      <ul>
        <li>
          <h3>
            dummy role @ dummy company - dummy city, dummy country
          </h3>
          <dl>
            <dt>
              Type:
            </dt>
            <dd>
              full-time
            </dd>
            <dt>
              Duration:
            </dt>
            <dd>
              10/2021 - 10/2022
            </dd>
          </dl>
          <div
            class="markdown"
          >
            <p>
              dummy contributions paragraph
            </p>
            

          </div>
        </li>
        <li>
          <h3>
            dummy role intern @ same company - another city, same country
          </h3>
          <dl>
            <dt>
              Type:
            </dt>
            <dd>
              part-time
            </dd>
            <dt>
              Duration:
            </dt>
            <dd>
              06/2021 - 09/2021
            </dd>
          </dl>
          <div
            class="markdown"
          >
            <p>
              dummy contributions paragraph as intern
            </p>
            

          </div>
        </li>
      </ul>
    </section>
    <section
      class="education"
      id="education"
    >
      <h2>
        Education
      </h2>
      <ul>
        <li>
          <h3>
            dummy school
          </h3>
          <p>
            BSc Dummy Career (2019 - 2021)
          </p>
          <dl>
            <dt>
              Cumulative GPA:
            </dt>
            <dd>
              90/100
            </dd>
            <dt
              class="subjects"
            >
              Most relevant subjects and courses:
            </dt>
            <dd>
              <ul
                class="ticks"
              >
                <li>
                  hello
                </li>
                <li>
                  world
                </li>
              </ul>
            </dd>
          </dl>
        </li>
        <li>
          <h3>
            dummy research centre
          </h3>
          <p>
            MSc Dummy Career (2021 - 2023)
          </p>
          <dl>
            <dt>
              Thesis:
            </dt>
            <dd>
              Dummy Thesis Topic
            </dd>
            <dt>
              Cumulative GPA:
            </dt>
            <dd>
              95/100
            </dd>
            <dt
              class="subjects"
            >
              Most relevant subjects and courses:
            </dt>
            <dd>
              <ul
                class="ticks"
              >
                <li>
                  hello master
                </li>
                <li>
                  world master
                </li>
              </ul>
            </dd>
          </dl>
        </li>
      </ul>
    </section>
    <section
      class="experiences"
      id="academic-projects"
    >
      <h2>
        Major Academic Projects
      </h2>
      <ul>
        <li>
          <h3>
            Dummy Project 01
          </h3>
          <dl>
            <dt>
              Type:
            </dt>
            <dd>
              thesis
            </dd>
            <dt>
              Duration:
            </dt>
            <dd>
              08/2013 - 11/2014
            </dd>
          </dl>
          <div
            class="markdown"
          >
            <p>
              Dummy description paragraph 01
            </p>
            

          </div>
        </li>
        <li>
          <h3>
            Dummy Project 02
          </h3>
          <dl>
            <dt>
              Type:
            </dt>
            <dd>
              personal
            </dd>
            <dt>
              Duration:
            </dt>
            <dd>
              Spring 2023
            </dd>
          </dl>
          <div
            class="markdown"
          >
            <p>
              Dummy description paragraph 02
            </p>
            

          </div>
        </li>
      </ul>
    </section>
    <section
      id="awards"
    >
      <h2>
        Awards
      </h2>
      <ul
        class="ticks"
      >
        <li>
          Dummy Award 01
        </li>
        <li>
          Dummy Award 02
        </li>
      </ul>
    </section>
    <section
      id="leadership"
    >
      <h2>
        Leadership
      </h2>
      <div
        class="markdown"
      >
        <p>
          Test leadership paragraph
        </p>
        

      </div>
    </section>
    <section
      id="volunteering"
    >
      <h2>
        Volunteering Activities
      </h2>
      <div
        class="markdown"
      >
        <p>
          Test volunteering paragraph
        </p>
        

      </div>
    </section>
  </div>
</DocumentFragment>
`;
